launch/model_endpoint.py

```diff
@@ -199,11 +199,18 @@ def __init__(self, client, endpoint_name: str, async_task_id: str):
         self.endpoint_name = endpoint_name
         self.async_task_id = async_task_id
 
-    def get(self) -> EndpointResponse:
+    def get(self, timeout: float = 0) -> EndpointResponse:
         """
         Retrieves the ``EndpointResponse`` for the prediction request after it completes. This method blocks.
+
+        Parameters:
+            timeout: The maximum number of seconds to wait for the response. If set to 0, then
+                the method will block indefinitely until the response is ready. If set to a positive
+                number, the method will raise a TimeoutError if the response is not ready after
+                ``timeout`` seconds.
         """
-        while True:
+        start_time = time.time()
+        while timeout == 0 or time.time() - start_time < timeout:
```

Comment:
should we do something if `timeout` is negative? (e.g. same behavior as timeout == 0 or something) comparing floats with equality seems a bit weird, even though we are comparing to 0 which should be fine in practice

Reply:
Added check 👍